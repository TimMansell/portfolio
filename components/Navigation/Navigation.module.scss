@import "styles/scssvars";
@import "styles/utilities/pxtorem";

.navigation {
  @media (min-width: $bp-lg) {
    position: sticky;
    top: 0;
    bottom: 0;
    height: pxtorem(60);
    background-color: rgba(#000, 0.7);
    z-index: var(--z-overlay);
  }

  &IsSticky {
    @media (min-width: $bp-lg) {
      box-shadow: inset 0 -1px var(--colour-1-alt-2), 0 1px 1px rgba(#000, 0.2);
      background-color: var(--colour-1-transparent);
    }
  }
}
