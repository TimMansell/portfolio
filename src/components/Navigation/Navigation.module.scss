@import 'variables';
@import 'utilities/pxtorem';

.navigation {
  @media (min-width: $bp-lg) {
    width: 100%;
    height: pxtorem(60);
    position: absolute;
    bottom: 0 !important;
    background-color: rgba(#000, 0.7);
    z-index: 999999;
  }

  &IsSticky {
    @media (min-width: $bp-lg) {
      position: fixed;
      top: 0 !important;
      box-shadow: inset 0 -1px var(--colour-1-alt-2), 0 1px 1px rgba(#000, 0.2);
      background-color: var(--colour-1-transparent);
    }
  }
}
