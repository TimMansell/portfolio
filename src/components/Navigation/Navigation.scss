@import 'variables';

.navigation {
  @media (min-width: $bp-lg) {
    width: 100%;
    height: 60px;
    position: absolute;
    bottom: 0 !important;
    z-index: 999999;
    background-color: rgba($tertiary-text-colour, 0.7);
  }

  &--active {
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 999999;
    background-color: rgba($tertiary-text-colour, 0.8);

    @media (min-width: $bp-lg) {
      position: relative;
    }
  }

  &--is-sticky {
    @media (min-width: $bp-lg) {
      position: fixed;
      top: 0 !important;
      box-shadow: inset 0 -1px darken($primary-colour, 10%), 0 1px 1px rgba(0, 0, 0, 0.2);
      background-color: rgba($primary-colour, 0.95);
    }
  }

  &__menu {
    display: none;
    text-align: center;
    list-style: none;
    margin: 0;

    @media (min-width: $bp-lg) {
      display: flex;
      justify-content: center;
    }

    &--active {
      display: block;
      position: fixed;
      width: 100%;
      bottom: 0;
      left: 50%;
      transform: translate(-50%, -100px);
      padding-left: 16px;
      padding-right: 16px;

      @media (min-width: $bp-lg) {
        display: none;
      }
    }
  }

  &__menu-item {
    @media (min-width: $bp-md) {
      line-height: 60px;
      text-transform: uppercase;
    }
  }

  &__menu-link {
    color: $secondary-text-colour;
    padding: 5px;
    display: block;
    font-size: 30px;
    letter-spacing: 2px;

    @media (min-width: $bp-sm) {
      padding: 1px;
    }

    @media (min-width: $bp-lg) {
      font-size: 18px;
      padding: 0 18px;
    }

    &--active,
    &:hover {
      @media (min-width: $bp-lg) {
        background-color: darken($primary-colour, 10%);
        box-shadow: inset 1px 0 darken($primary-colour, 11%), inset -1px 0 darken($primary-colour, 11%);
      }
    }

    &--is-sticky {
      @media (min-width: $bp-lg) {
        text-shadow: -1px -1px darken($primary-colour, 10%);
      }
    }
  }
}
