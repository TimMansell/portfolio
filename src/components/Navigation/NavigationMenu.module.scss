@import 'variables';
@import 'utilities/pxtorem';

.menu {
  display: none;

  @media (min-width: $bp-lg) {
    display: flex;
    justify-content: center;
    height: 100%;
  }

  &IsActive {
    display: flex;
    flex-direction: column;
    position: fixed;
    width: 100%;
    bottom: 0;
    transform: translateY(-100px);
    padding-left: var(--vr);
    padding-right: var(--vr);
    z-index: var(--z-overlay);

    @media (orientation: landscape) {
      flex-direction: row;
      flex-wrap: wrap;
      justify-content: center;
      gap: var(--vr);
      padding-left: var(--vr-md);
      padding-right: var(--vr-md);
    }

    @media (min-width: $bp-lg) {
      display: none;
    }
  }
}

.menuItem {
  display: flex;
  justify-content: center;

  @media (min-width: $bp-md) {
    text-transform: uppercase;
  }
}

.link {
  display: flex;
  align-items: center;
  color: var(--secondary-text-colour);
  padding: var(--vr-xxs);
  font-size: pxtorem(30);
  letter-spacing: pxtorem(2);

  @media (orientation: landscape) {
    font-size: pxtorem(22);
    padding: var(--vr-xs);
    border: 1px solid rgba(255, 255, 255, 0.15);
    border-radius: 7px;
  }

  @media (min-width: $bp-lg) {
    font-size: pxtorem(16);
    padding: 0 var(--vr-xs);
    border: 0;
    border-radius: 0;
  }

  @media (min-width: $bp-xl) {
    font-size: pxtorem(18);
    padding: 0 var(--vr);
  }

  &IsActive,
  &:hover {
    @media (min-width: $bp-lg) {
      background-color: var(--colour-1-alt-2);
      box-shadow:
        inset 1px 0 var(--colour-1-alt-2),
        inset -1px 0 var(--colour-1-alt-2);
    }
  }

  &IsSticky {
    @media (min-width: $bp-lg) {
      text-shadow: -1px -1px var(--colour-1-alt-2);
    }
  }
}
