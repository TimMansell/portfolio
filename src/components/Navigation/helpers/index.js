import {
  combineWords,
  splitWords,
  upperCaseWord,
  upperCaseWords,
} from './helpers';

export { combineWords, splitWords, upperCaseWord, upperCaseWords };
