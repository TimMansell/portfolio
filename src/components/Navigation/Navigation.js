import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { setMobileMenu } from 'actions';

import classnames from 'classnames';
import { Link } from 'react-scroll';

import Hamburger from 'components/Hamburger';

import './Navigation.scss';

export class Navigation extends React.Component {
  constructor (props) {
    super(props);

    this.state = {
      isFixedNav: false
    };

    this.refNavigation = React.createRef();
  }

  componentDidMount () {
    window.addEventListener('scroll', this.handleScroll);

    this.setState(prevState => ({
      offset: this.refNavigation.current.offsetTop
    }));
  }

  componentWillUnmount () {
    window.removeEventListener('scroll', this.handleScroll);
  }

  handleScroll = (event) => {
    const scrollTop = window.scrollY || document.documentElement.scrollTop;
    const { offset } = this.state;
    // eslint-disable-next-line no-tabs
    let	isFixedNav = false;

    if (scrollTop >= offset) {
      isFixedNav = true;
    } else {
      isFixedNav = false;
    }

    this.setState(prevState => ({
      isFixedNav: isFixedNav
    }));
  }

  closeMenu = () => {
    const { setMobileMenu } = this.props;

    setMobileMenu(false);
  }

  render () {
    const { isMobileMenu } = this.props;
    const { isFixedNav } = this.state;

    const navClasses = classnames('navigation', {
      'navigation--is-sticky': isFixedNav && !isMobileMenu,
      'navigation--active': isMobileMenu
    });

    const menuClasses = classnames('navigation__menu', {
      'navigation__menu--active': isMobileMenu
    });

    const linkClasses = classnames('navigation__menu-link', {
      'navigation__menu-link--is-sticky': isFixedNav && !isMobileMenu
    });

    return <nav id="nav" className={navClasses} ref={this.refNavigation}>
      <ul className={menuClasses}>
        <li className="navigation__menu-item">
          <Link data-e2e="navigation-profile" className={linkClasses} activeClass="navigation__menu-link--active" to="profile" smooth={true} spy={true} duration={500} onClick={this.closeMenu}>Profile</Link>
        </li>
        <li className="navigation__menu-item">
          <Link data-e2e="navigation-skills" className={linkClasses} activeClass="navigation__menu-link--active" to="skills" smooth={true} spy={true} duration={500} onClick={this.closeMenu}>Skills</Link>
        </li>
        <li className="navigation__menu-item">
          <Link data-e2e="navigation-retired-skills" className={linkClasses} activeClass="navigation__menu-link--active" to="retired-skills" smooth={true} spy={true} duration={500} onClick={this.closeMenu}>Retired Skills</Link>
        </li>
        <li className="navigation__menu-item">
          <Link data-e2e="navigation-stack" className={linkClasses} activeClass="navigation__menu-link--active" to="stack" smooth={true} spy={true} duration={500} onClick={this.closeMenu}>Stack</Link>
        </li>
        <li className="navigation__menu-item">
          <Link data-e2e="navigation-portfolio" className={linkClasses} activeClass="navigation__menu-link--active" to="portfolio" smooth={true} spy={true} duration={500} onClick={this.closeMenu}>Portfolio</Link>
        </li>
        <li className="navigation__menu-item">
          <Link data-e2e="navigation-presentations" className={linkClasses} activeClass="navigation__menu-link--active" to="presentations" smooth={true} spy={true} duration={500} onClick={this.closeMenu}>Presentations</Link>
        </li>
        <li className="navigation__menu-item">
          <Link data-e2e="navigation-testimonials" className={linkClasses} activeClass="navigation__menu-link--active" to="testimonials" smooth={true} spy={true} duration={500} onClick={this.closeMenu}>Testimonials</Link>
        </li>
        <li className="navigation__menu-item">
          <Link data-e2e="navigation-contact" className={linkClasses} activeClass="navigation__menu-link--active" to="contact" smooth={true} spy={true} duration={500} onClick={this.closeMenu}>Contact</Link>
        </li>
      </ul>

      <Hamburger />
    </nav>;
  }
}

function mapStateToProps (state, ownProps) {
  return {
    isMobileMenu: state.isMobileMenu
  };
}

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({ setMobileMenu }, dispatch);
};

Navigation.propTypes = {
  isMobileMenu: PropTypes.bool,
  setMobileMenu: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(Navigation);
