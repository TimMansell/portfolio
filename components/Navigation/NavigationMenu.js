import React, {useContext} from "react";
import {MenuContext} from "../../context/mobileMenu";
import classnames from "classnames";
import {Link} from "react-scroll";

import useInMobileViewport from "hooks/useInMobileViewport";
import styles from "./NavigationMenu.module.scss";
import navigationJson from "./json/navigation.json";

import {combineWords, upperCaseWords} from "./helpers";

import {useIsFlexGapSupported} from "hooks/useFlexGap";

export const NavigationMenu = () => {
  const hasFlexGap = useIsFlexGapSupported();
  const [isMobileMenu, setIsMobileMenu] = useContext(MenuContext);
  const inViewport = useInMobileViewport();

  const menuClasses = classnames(styles.menu, {
    [styles.menuIsActive]: isMobileMenu && inViewport,
    [styles.flexGap]: hasFlexGap,
  });

  return (
    <ul className={menuClasses} data-e2e="navigation">
      {navigationJson.map((navigation, index) => (
        <li className={styles.menuItem} key={index}>
          <Link
            data-e2e={`navigation-${combineWords(navigation)}`}
            className={styles.link}
            activeClass={styles.linkIsActive}
            to={combineWords(navigation)}
            smooth={true}
            spy={true}
            duration={500}
            onClick={() => setIsMobileMenu(false)}
          >
            {upperCaseWords(navigation)}
          </Link>
        </li>
      ))}
    </ul>
  );
};

export default NavigationMenu;
