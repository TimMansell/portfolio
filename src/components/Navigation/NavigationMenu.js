import React, { useContext } from 'react';
import { MenuContext } from '../../context/mobileMenu';
import classnames from 'classnames';
import { Link } from 'react-scroll';

import useInMobileViewport from 'hooks/useInMobileViewport';
import styles from './NavigationMenu.module.scss';
import navigationJson from './json/navigation.json';

import { combineWords, upperCaseWords } from './helpers';

export const NavigationMenu = () => {
  const [isMobileMenu, setIsMobileMenu] = useContext(MenuContext);
  const inViewport = useInMobileViewport();

  const menuClasses = classnames(styles.menu, {
    [styles.menuIsActive]: isMobileMenu && inViewport,
  });

  return (
    <ul className={menuClasses}>
      {navigationJson.map((navigation, index) => (
        <li className={styles.menuItem} key={index}>
          <Link
            data-e2e={`navigation-${combineWords(navigation)}`}
            className={styles.link}
            activeClass={styles.linkIsActive}
            to={combineWords(navigation)}
            smooth={true}
            spy={true}
            duration={500}
            onClick={() => setIsMobileMenu(false)}
          >
            {upperCaseWords(navigation)}
          </Link>
        </li>
      ))}
    </ul>
  );
};

export default NavigationMenu;
