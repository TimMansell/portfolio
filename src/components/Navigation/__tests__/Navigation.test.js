import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { Navigation } from '../Navigation';

describe("Navigation", () => {
  it("should render my component", () => {
    const wrapper = shallow(<Navigation />, {disableLifecycleMethods:true});
  });

  it.skip("should match snapshot", () => {
    const snapshot = renderer.create(<Navigation/>).toJSON();
    
    expect(snapshot).toMatchSnapshot();
  });
});
