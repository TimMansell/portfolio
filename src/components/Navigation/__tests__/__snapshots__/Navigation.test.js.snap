// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navigation"
    id="nav"
    role="navigation"
  >
    <ul
      class="menu"
      data-e2e="navigation"
    >
      <li
        class="menuItem"
      >
        <a
          class="link"
          data-e2e="navigation-profile"
        >
          Profile
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="link"
          data-e2e="navigation-skills"
        >
          Skills
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="link"
          data-e2e="navigation-older-skills"
        >
          Older Skills
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="link"
          data-e2e="navigation-stack"
        >
          Stack
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="link"
          data-e2e="navigation-portfolio"
        >
          Portfolio
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="link"
          data-e2e="navigation-testimonials"
        >
          Testimonials
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="link"
          data-e2e="navigation-contact"
        >
          Contact
        </a>
      </li>
    </ul>
    <div
      class="container"
    >
      <button
        class="hamburger"
        data-e2e="hambuger"
        type="button"
      >
        <span
          class="menu"
        >
          toggle menu
        </span>
      </button>
      <div
        class="background"
      />
    </div>
  </nav>
</DocumentFragment>
`;
